subscriptions/fixed-price/checkout/node/src/server.ts

```diff
@@ -207,3 +247,248 @@ app.post('/webhook', async (req, res) => {
+        // ...
+
+        return
+      ```
+    */
+  }
+  switch (charge.reason) {
+    case 'subscription_creation': // S1 ✅
+    case 'subscription_due': // S2 ✅
+    case 'subscription_update': // S4 ✅
+    case 'subscription_cancel': // S5 ✅
+      /*
+        You may want to do the following:
+        - Send an invoice to your customer.
+        - Store charges in your DB.
+      */
+      break
+
+    default:
+      break
+  }
+}
+
+/**
+ * Subscription created
+ *
+ * We recommend pivoting on `subscription.created`, to handle the case when a checkout is a completed.
+ * Use this event to provide feedback though the UI, and create a new subscription in your database.
+ *
+ * @param subscription The subscription object received in the event
+ */
+async function handleSubscriptionCreated(
+  subscription: DiagonalSubscription,
+): Promise<void> {
+  /*
+      Find the user in your database using the `customer_id` or `reference`, specified on the subscription object:
+
+      ```
+        const user = UserTable.findOne({
+          diagonalCustomerId: subscription.customer_id!,
+        })
+      ```
+
+      Create a new subscription in your database:
+    
+      ```
+        SubscriptionTable.create({
+          userId: user.id,
+          status: 'active',
+          diagonalSubscriptionId: subscription.id,
+          planId: subscription.reference,
+        })
+      ```
+
+  */
+}
+
+/**
+ * Charge attempt failed
+ *
+ * @param charge The charge object received in the event
+ */
+async function handleChargeAttemptFailed(charge: Charge): Promise<void> {
+  /*
+    You may want to notify your user that the subscription charge failed, 
+    based on the failure reason specified in `charge.last_attempt_failure_reason`
+
+    NOTE: charge will be re-attempted at `charge.next_attempt_at`
+  */
+
+  switch (charge.last_attempt_failure_reason) {
+    case 'insufficient_allowance':
+      // Notify the user to increase their spending allowance on subscriptions.diagonal.finance
+      break
+    case 'insufficient_balance':
+      // Notify the user to fund their wallet
+      break
+    default:
+      break
+  }
+}
+
+/**
+ * Charge failed
+ *
+ *  We recommend pivoting on `charge.failed` when handling the following subscription lifecycle events:
+ *
+ *   - S1: Subscription creation failed
+ *
+ *  DISCLAIMER:
+ *  `charge.failed` can fire in other scenarios e.g. "maximum number of retry attempts reached" or "address blacklist usdc".
+ *  We recommend handling these scenarios in `handleSubscriptionCanceled`.
+ *
+ * @param charge The charge object received in the event
+ */
+async function handleChargeFailed(charge: Charge): Promise<void> {
+  if (charge.reason === 'subscription_creation') {
+    /*
+      Handle subscription creation failed (S1 ✅)
+
+      1: Remove diagonal subscription entry from your database
+      ```
+        SubscriptionTable.deleteOne({ diagonalSubscriptionId: charge.subscription_id })
```

Comment:
Probably instead of remove, we can update the status of the subscription to Paused locally...

Reply:
Actually, at this stage this is OK, and whenever/if we support resubscribing to the same subscription, then we can update this.